va/va.h

```diff
@@ -4225,6 +4225,62 @@ VAStatus vaDeriveImage (
     VAImage *image	/* out */
 );
 
+typedef struct _VASurfaceAllocationInfo{
+    /* format fourcc */
+    uint32_t    fourcc;
+    /* picture width x height */
+    uint16_t	width; 
+    uint16_t	height;
+    /* suggest data_size, it alwsys is minimum requirement*/ 
+    uint32_t	data_size;
+    /* number of planes , can not be greater than 3 */
+    uint32_t	num_planes;
+    /* 
+     * An array indicating the scanline pitch in bytes for each plane.
+     * Each plane may have a different pitch. Maximum 3 planes for planar formats
+     */
+    uint32_t	pitches[4];
+    /* 
+     * An array indicating the byte offset from the beginning of the image data 
+     * to the start of each plane.
+     */
+    uint32_t	offsets[4];
+    /*
+     * An array of ascii characters describing the order of the components within the bytes.
+     * Only entry_bytes characters of the string are used.by default , the order is from format fourcc
+     * if there are different layout, can use this value to describe it.
+     */
+    int8_t component_order[4];
```

Comment:
do we actually need component_order? Isn't it assumed by the color format? Also, VASurfaceAttribExternalBuffers does not have it, so it's a question how user will be able to use component_order info?

Reply:
if we remove it, will specify the order, fro example: YUV444 planar, but maybe some application use different buffer to store each plane, specify it should follow Y, U, V order.